geolocator_android/lib/src/types/foreground_settings.dart

```diff
@@ -0,0 +1,74 @@
+/// Uniquely identifies an Android resource
+class AndroidResource {
+  /// The name of the desired resource.
+  final String name;
+
+  /// Optional default resource type to find, if "type/" is not included in the name. Can be null to require an explicit type.
+  final String defType;
+
+  /// Uniquely identifies an Android resource
+  const AndroidResource({required this.name, this.defType = 'drawable'});
+
+  /// Returns a json representation of this class
+  Map<String, dynamic> toJson() {
+    return {
+      'name': name,
+      'defType': defType,
+    };
+  }
+}
+
+/// Configuration for the foreground notification. When this is provided the location service will run as a foreground service.
+class ForegroundNotificationConfig {
+  /// The title used for the foreground service notification.
+  final String notificationTitle;
+
+  /// The body used for the foreground service notification.
+  final String notificationText;
+
+  /// The resource name of the icon to be used for the foreground notification.
+  final AndroidResource notificationIcon;
+
+  /// When enabled, a WifiLock is acquired when background execution is started.
+  /// This allows the application to keep the Wi-Fi radio awake, even when the
+  /// user has not used the device in a while (e.g. for background network
+  /// communications).
+  ///
+  /// Wifilock permissions should be obtained first by using a permissions library
```

Comment:
Nit: sentence should end with a period.
```suggestion
  /// Wifi lock permissions should be obtained first by using a permissions library.
```